media-tv/kodi/kodi-20.2.ebuild

```diff
@@ -0,0 +1,462 @@
+		-DENABLE_INTERNAL_KISSFFT=OFF
+		-DENABLE_INTERNAL_PCRE=OFF
+		-DENABLE_INTERNAL_RapidJSON=OFF
+		-DENABLE_INTERNAL_SPDLOG=OFF
+		-DENABLE_INTERNAL_TAGLIB=OFF
+		-DENABLE_INTERNAL_UDFREAD=OFF
+
+		-DWITH_FFMPEG=$(usex system-ffmpeg)
+
+		-Dlibdvdnav_URL="${DISTDIR}/libdvdnav-${LIBDVDNAV_VERSION}.tar.gz"
+		-Dlibdvdread_URL="${DISTDIR}/libdvdread-${LIBDVDREAD_VERSION}.tar.gz"
+	)
+
+	# Separated to avoid "Manually-specified variables were not used by the project:"
+	use css && mycmakeargs+=( -Dlibdvdcss_URL="${DISTDIR}/libdvdcss-${LIBDVDCSS_VERSION}.tar.gz" )
+	use !system-ffmpeg && mycmakeargs+=(
+		-DFFMPEG_URL="${DISTDIR}/ffmpeg-${PN}-${FFMPEG_VERSION}-${CODENAME}-${FFMPEG_KODI_VERSION}.tar.gz"
+	)
+	use nfs && mycmakeargs+=( -DENABLE_INTERNAL_NFS=OFF )
+	use !udev && mycmakeargs+=( -ENABLE_LIBUSB=$(usex libusb) )
+	use X && use !gles && mycmakeargs+=( -DENABLE_GLX=ON )
+
+	for flag in ${CPU_FLAGS[@]} ; do
+		local name=${flag#cpu_flags_*_}
+		mycmakeargs+=( -DENABLE_${name^^}=$(usex ${flag}) )
+	done
+
+	if ! is-flag -DNDEBUG && ! is-flag -D_DEBUG ; then
+		# Kodi requires one of the 'NDEBUG' or '_DEBUG' defines
+		append-cflags -DNDEBUG
+		append-cxxflags -DNDEBUG
+	fi
+
+	# Violates ODR (bug #860984) and USE_LTO does spooky stuff
+	# https://github.com/xbmc/xbmc/commit/cb72a22d54a91845b1092c295f84eeb48328921e
+	filter-lto
+
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile all
+	use doc && cmake_build doc
+	use test && cmake_build kodi-test
+}
+
+src_test() {
+	local -x CMAKE_SKIP_TESTS=(
+		# Known failing, unreliable test
+		# bug #743938
+		TestCPUInfo.GetCPUFrequency
+		# Test failure stemming from sci-libs/kissfft
+		# The difference between output[2i] and (i==freq1?1.0:0.0) is inf, which exceeds 1e-7, where output[2i]
+		# evaluates to inf,(i==freq1?1.0:0.0) evaluates to 0, and 1e-7 evaluates to 9.9999999999999995e-08.
+		TestRFFT.SimpleSignal
+		# bug #779184
+		# https://github.com/xbmc/xbmc/issues/18594
+		$(usev x86 TestDateTime.SetFromDBTime)
+	)
+
+	# see https://github.com/xbmc/xbmc/issues/17860#issuecomment-630120213
+	local -x KODI_HOME="${BUILD_DIR}"
+
+	cmake_src_test
+}
+
+src_install() {
+	cmake_src_install
+
+	# bug #457588
+	pax-mark Em "${ED}"/usr/$(get_libdir)/${PN}/${PN}.bin
+
+	newicon media/icon48x48.png kodi.png
+
+	rm "${ED}"/usr/share/kodi/addons/skin.estuary/fonts/Roboto-Thin.ttf || die
+	dosym ../../../../fonts/roboto/Roboto-Thin.ttf \
+		usr/share/kodi/addons/skin.estuary/fonts/Roboto-Thin.ttf
+
+	if use !eventclients ; then
+		rm -f "${ED}"/usr/bin/kodi-ps3remote || die
+		rm -f "${D}"$(python_get_sitedir)/kodi/ps3_remote.py || die
+		rm -rf "${D}"$(python_get_sitedir)/kodi/ps3 || die
+		rm -rf "${D}"$(python_get_sitedir)/kodi/bt || die
+		rm -rf "${ED}"/usr/share/doc/${PF}/kodi-eventclients-dev || die
+	fi
+
+	python_optimize "${D}$(python_get_sitedir)"
+
+	einstalldocs
+	use doc && dodoc -r "${S}"/docs/html/
+}
+
+pkg_postinst() {
+	xdg_pkg_postinst
+
+	if use dbus ; then
```

Comment:
Maybe better with just a optfeature_header saying that these are dependencies that require dbus?